lib/AST/Decl.cpp

```diff
@@ -5625,6 +5625,53 @@ bool VarDecl::isPropertyMemberwiseInitializedWithWrappedType() const {
       this, /*checkDefaultInit=*/false);
 }
 
+static bool isInitFirstParamEscapingClosure(const ConstructorDecl *init) {
+  Type initTy = init->getInterfaceType();
+  if (!initTy->is<AnyFunctionType>())
+    return false;
+
+  Type resultTy = initTy->castTo<AnyFunctionType>()->getResult();
```

Comment:
Just a minor stylistic thing: rather than `->is<AnyFunctionType>()` followed by `->castTo<AnyFunctionType>()`, you can use `->getAs<AnyFunctionType>()` and then check if the resulting pointer is NULL.